scripts/simplify_network.py

```diff
@@ -354,37 +375,80 @@ def split_links(nodes):
                 n, busmap, costs, config, connection_costs_per_link, buses
             )
 
-            all_links = [i for _, i in sum(links, [])]
+            # TODO revise a variable name for `links` variable
+            # `links` is a list which contains dc-relevant graph elements
+            all_dc = [i for _, i in sum(links, [])]
+            all_links = list(set(n.links.index).intersection(all_dc))
+            all_dc_lines = list(set(n.lines.index).intersection(all_dc))
```

Comment:
Sorry Katia, my bad, but I don't understand this line: you are making the intersection of the line index with all_dc that are indices of links, right?

I see above that links is an object that comes out of graph analyses, may it be worth to rename it to like all_dc_components or something like that?
May also be worth checking if it contains the expected objects of both links and lines, but I guess you have already checked that.

Amazing! :D

Reply:
Yes, absolutely: here I'm trying to split `links` list on `Line` and `Link` components. If it's not clear it means that this fragment should be definitely improved.

Have revised naming changing `all_dc` -> `all_dc_branches` and added a bit to the comments.

Not sure I have get your point regarding the checking which does likely mean that I miss something... Would be very grateful for a short exchange on that :)

Reply:
Indeed, there is [a check](https://github.com/ekatef/pypsa-earth/blob/50b558f091efd8ebb87ef76fef90228382be1d92/scripts/simplify_network.py#L278-L281) above in `contains_ac()`, but I'm not sure that is what you mean

Reply:
My concern was mainly in the link list, doesn't it contains also line branches? I feel like its name is misleading, what do you think?

Reply:
Absolutely agree. This naming has been inherited from PyPSA-Eur, and I have tried to keep it. Happy to revise :)

Reply:
Have tired to improve naming. Not completely sure if it's worth. Happy to undo in case :)